packages/react/src/number-field/utils/parse.ts

```diff
@@ -46,9 +46,15 @@ export function parseNumber(
   }
 
   const { group, decimal, currency, unit } = getNumberLocaleDetails(computedLocale, options);
+  let groupRegex: RegExp | null = null;
+  if (group) {
+    // Check if the group separator is a space-like character.
+    // If so, we'll replace all such characters with an empty string.
+    groupRegex = /\p{Zs}/u.test(group ?? '') ? /\p{Zs}/gu : new RegExp(`\\${group}`, 'g');
+  }
```

Comment:
`if (group)` and `group ?? ''` seems redundant?

Reply:
Copy-pasta...